miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleRestController.java

```diff
@@ -363,6 +372,84 @@ protected Sample fetch(long id) throws IOException {
     return parentFinder.list(ids, category);
   }
 
+  private final RelationFinder<Sample> childFinder = (new RelationFinder<Sample>() {
+
+    @Override
+    protected Sample fetch(long id) throws IOException {
+      return sampleService.get(id);
+    }
+  })
+      .add(RelationFinder.child(SampleIdentity.CATEGORY_NAME, SampleIdentity.class))//
+      .add(RelationFinder.child(SampleTissue.CATEGORY_NAME, SampleTissue.class))//
+      .add(RelationFinder.child(SampleTissueProcessing.CATEGORY_NAME, SampleTissueProcessing.class))//
+      .add(RelationFinder.child(SampleStock.CATEGORY_NAME, SampleStock.class))//
+      .add(RelationFinder.child(SampleAliquot.CATEGORY_NAME, SampleAliquot.class))
+      
+      .add(new RelationFinder.RelationAdapter<Sample, Library, LibraryDto>("Library") {
+
+        @Override
+        public LibraryDto asDto(Library model) {
+          return Dtos.asDto(model);
+        }
+
+        @Override
+        public Stream<Library> find(Sample model, Consumer<String> emitError) {
+          Set<Library> children = RelationFinder.ChildrenSampleAdapter.searchChildrenLibraries((DetailedSample) model)
+              .collect(Collectors.toSet());
+          if (children.isEmpty()) {
+            emitError.accept(String.format("%s (%s) has no %s.", model.getName(), model.getAlias(), category()));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "%s (%s) has no %s." 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)